test/steps/application_v2_steps.py

```diff
@@ -512,35 +537,37 @@ def add_transaction_to_composer(context):
     )
 
 
-def process_abi_args(context, method, arg_tokens):
+def process_abi_args(context, method, arg_tokens, erase_empty_tokens=True):
+    if erase_empty_tokens:
```

Comment:
Now that we no longer assume that the lengths of the method args and actual args are the same, a couple of side effects occurred. One side effect is that when `"".split()` is called on an empty string, you get a non-empty list. As in all existing cases, empty strings that are split meant to indicate no values at all, this change shouldn't break any tests.

Reply:
I'm a little confused. Is the cause of this change that the step `I append the encoded arguments "<app-args>" to the method arguments array` now has to work when `<app-args>` is the empty string?

Or is there another cause?

Reply:
Previously, all tests had exactly the same number of runtime args as defined in the method signature. The step logic iterated through the _method signature_ args and accessed the run-time arg with index corresponding to the signature arg. When there were no _method signature_ args, the loop was actually trivial (see the image below). However, in that trivial case, `args` was actually the list of length 1 containing an empty string `['']` (because of how python split works):
```python
>>> "".split(",")
['']
```
Now that we are no longer assuming that the number of args are the same, the 0 vs. 1 case no longer was correct so I needed to adjust the logic using this hack.

<img width="473" alt="image" src="https://user-images.githubusercontent.com/291133/174354271-ea9ca974-b23d-4d51-bce9-0d053d26e0e6.png">

Reply:
So yes, it has to work with empty string (which before was actually ignored in a sense).

Reply:
Thanks for that explanation, I think I understand now. Alternatively, would it be an acceptable solution to modify the `I append the encoded arguments "<app-args>" to the method arguments array` step to check if `<app-args>` is an empty string and do nothing in that case?

The reason I suggest this is because we can semantically still pass empty string args with something like `a,,b`, and this would be ignored by the current code, but I think it should complain/error for this input.

Reply:
yeah, you'd have to set `erase_empty_tokens` to false... still thinking about your first paragraph

Reply:
I have a proposal. Rename this step to:

```
'I add a method call with the {account_type} account, the current application, suggested params, on complete "{operation}", current transaction signer, current method arguments ignoring empty ones; any resulting exception has key "{exception_key}".'
```
the phrase "ignoring empty ones" indicates that we should not be allowing empty arguments for this step. And in fact, this step will eventually call `process_abi_args(... erase_empty_tokens=True)` while all the other steps will continue as before and therefor have `erase_empty_tokens=False`

Sounds good?

Reply:
That's better, but it still looks like we're addressing the symptoms of the problem, not the root cause. I'd find it simpler if no filtering of arguments needed to be done after the fact, since this seems error-prone and hard to reason about.

Reply:
I'm trying your original suggestion with `I append the encoded arguments`. Seems to work. We'll see if the tests pass.